packages/transcription/src/whisper/transcriber/ctranslate2-transcriber.ts

```diff
@@ -1,47 +1,49 @@
 import { $ } from 'execa'
-import short, { SUUID } from 'short-uuid'
-import { join } from 'path'
+import { buildSUUID } from '@peertube/peertube-node-utils'
 import { lstat } from 'node:fs/promises'
-import { OpenaiTranscriber } from './openai-transcriber.js'
-import { TranscriptionModel } from '../../transcription-model.js'
-import { TranscriptFile, TranscriptFormat } from '../../transcript/index.js'
-import { getFileInfo } from '../../file-utils.js'
+import { OpenaiTranscriber, WhisperTranscribeArgs } from './openai-transcriber.js'
+import { TranscriptFile } from '../../transcript/index.js'
+import { WhisperBuiltinModel } from '../whisper-builtin-model.js'
 
 export class Ctranslate2Transcriber extends OpenaiTranscriber {
-  async transcribe (
-    mediaFilePath: string,
-    model: TranscriptionModel = { name: 'tiny' },
-    language: string = 'en',
-    format: TranscriptFormat = 'vtt',
-    runId: SUUID = short.generate()
-  ): Promise<TranscriptFile> {
+  async transcribe ({
+    mediaFilePath,
+    model = new WhisperBuiltinModel('tiny'),
+    language,
+    format = 'vtt',
+    runId = buildSUUID()
+  }: WhisperTranscribeArgs): Promise<TranscriptFile> {
+    this.assertLanguageDetectionAvailable(language)
+
     // Shall we run the command with `{ shell: true }` to get the same error as in sh ?
     // ex: ENOENT => Command not found
     const $$ = $({ verbose: true })
-    const { baseName } = getFileInfo(mediaFilePath)
 
     if (model.path) {
       await lstat(model.path).then(stats => stats.isDirectory())
```

Comment:
Not sure to understand what's the point of this line?

Reply:
You're right, let's turn this into an assertion.